src/form.js

```diff
@@ -897,14 +907,27 @@ define([
                 mug.options.afterInsert(this, mug);
             }
         },
-        _updateMugPath: function (mug, oldPath, newPath) {
-            var map = this.mugMap;
-            delete map[oldPath];
-            if (_.isUndefined(newPath)) {
+        _updateMugPath: function (mug, oldHashtag, newHashtag) {
+            var map = this.mugMap, newPath;
+            delete map[normalizeHashtag(oldHashtag)];
+            if (oldHashtag) {
+                xpath.removeHashtag(oldHashtag);
```

Comment:
`removeHashtag` should also be a method on the form.

Actually, maybe it should remain a method of the `xpath` object, but the `xpath` object should be created and owned by the form rather than being a global. That's a better idea than what I outlined above because it encapsulates the details of creating an xpath parser within the `xpath` module rather than bloating the form with those details + methods. If you go that route then this call will become `form.xpath.removeHashtag(oldHashtag)`